modin/experimental/engines/omnisci_on_ray/io.py

```diff
@@ -456,13 +457,24 @@ def _read_csv_check_support(
                 "Specified a delimiter with both sep and delim_whitespace=True; you can only specify one."
             )
 
+        parse_dates_unsupported = isinstance(parse_dates, dict) or (
+            isinstance(parse_dates, list) and isinstance(parse_dates[0], list)
+        )
+        if parse_dates_unsupported:
+            return (
+                False,
+                "read_csv with 'arrow' engine supports only bool and "
+                "flattened lists `parse_dates` parameter",
+            )
         if names:
             if header not in [None, 0, "infer"]:
                 return (
                     False,
                     "read_csv with 'arrow' engine and provided 'names' parameter supports only 0, None and "
                     "'infer' header values",
                 )
+            if isinstance(parse_dates, list) and not set(parse_dates).issubset(names):
+                raise ValueError("Non-existent column provided to 'parse_dates'")
```

Comment:
what exception does pandas report in this case?

Reply:
In this case [pandas raises](https://github.com/pandas-dev/pandas/blob/1.2.x/pandas/io/parsers.py#L1415-L1417) `ValueError` with message `f"Missing column provided to 'parse_dates': '{missing_cols}'"`.

Reply:
Uh-oh, in this case we better revert to pandas messaging to be consistent... sorry about the noise 😄

Reply:
Ok, changes were reverted)